### PR TITLE
Add gradedBy to Answers (Resolves #260)

### LINKED ISSUE #260: /grading should return the grader, if any.
The grader is the staff member who approves the grade.

This will allow front end to determine if the problem/mission has already been graded.

Simultaneously, there should be an endpoint to "confirm" the grading (i.e. for acknowledging autograding).

Recommend implementation by adding a user_id column to the table.

### PR DESCRIPTION
**Pull request by jiachen247**
Resolves #260

Added a new field to the database to track grading by avengers.

Note that this only tracks the latest avenger to grade the assessment. 

If assessment has not been graded by an avenger, null is returned.

### CODE REVIEW

**Comment by jiachen247** [comment]
shall we expose a `gradedAt`param as well to reflect when it was last graded?

File: lib/cadet/assessments/assessments.ex [hunk 1]
```diff
@@ -145,7 +145,8 @@ defmodule Cadet.Assessments do
         Question
         |> where(assessment_id: ^id)
         |> join(:left, [q], a in subquery(answer_query), q.id == a.question_id)
-        |> select([q, a], %{q | answer: a})
+        |> join(:left, [q, a], g in assoc(a, :grader))
```
Comment (indocomsoft) [inline lib/cadet/assessments/assessments.ex]: Prefer `[_, a]` since `q` is not used

File: lib/cadet/assessments/assessments.ex [hunk 2]
```diff
@@ -524,10 +525,12 @@ defmodule Cadet.Assessments do
   def update_grading_info(
         %{submission_id: submission_id, question_id: question_id},
         attrs,
-        grader = %User{role: role}
+        grader = %User{id: grader_id, role: role}
       )
       when is_ecto_id(submission_id) and is_ecto_id(question_id) and
              (role in @grading_roles or role in @see_all_submissions_roles) do
+    attrs = Map.put(attrs, "grader_id", grader_id)
```
Comment (indocomsoft) [inline lib/cadet/assessments/assessments.ex]: Why not use `:grader_id` as key?

File: lib/cadet_web/controllers/assessments_controller.ex [hunk 3]
```diff
@@ -187,7 +187,9 @@ defmodule CadetWeb.AssessmentsController do
               "The library used for this question"
             )
 
-            solution_template(:string, "Solution template for programming questions")
+            solutionTemplate(:string, "Solution template for programming questions")
```
Comment (indocomsoft) [inline lib/cadet_web/controllers/assessments_controller.ex]: Frontend will updated for this right?

File: lib/cadet_web/controllers/assessments_controller.ex [hunk 4]
```diff
@@ -187,7 +187,9 @@ defmodule CadetWeb.AssessmentsController do
...
+            
+            gradedBy(:string, "Name of Avenger that last graded the answer.")
```
Comment (indocomsoft) [inline lib/cadet_web/controllers/assessments_controller.ex]: who instead of that? haha

File: lib/cadet_web/views/assessments_view.ex [hunk 5]
```diff
@@ -123,6 +123,7 @@ defmodule CadetWeb.AssessmentsView do
     transform_map_for_view(answer, %{
       answer: answer_getter,
       comment: :comment,
+      gradedBy: & &1.grader[:name],
```
Comment (indocomsoft) [inline lib/cadet_web/views/assessments_view.ex]: You need to make sure to preload `grader` in `Assessments.assessment_with_questions_and_answers/2`

File: lib/cadet_web/controllers/assessments_controller.ex (hunk 3 above)
Comment (jiachen247) [reply lib/cadet_web/controllers/assessments_controller.ex]: actually both front and backend are using camel case already. The doc was outdated.

File: lib/cadet_web/views/assessments_view.ex (hunk 5 above)
Comment (jiachen247) [reply lib/cadet_web/views/assessments_view.ex]: already being preloaded.

File: lib/cadet/assessments/assessments.ex (hunk 2 above)
Comment (jiachen247) [reply lib/cadet/assessments/assessments.ex]: all keys in attr are strings not atoms? It throws this error when i try to use an atom instead.

```
 ** (Ecto.CastError) expected params to be a map with atoms or string keys, got a map with mixed keys: %{:grader_id => 48101, "adjustment" => -100, "comment" => "Your awesome", "xpAdjustment" => -100, "xp_adjustment" => -100}

```

Comment (indocomsoft) [reply lib/cadet/assessments/assessments.ex]: Right, my bad.

File: lib/cadet_web/controllers/assessments_controller.ex [hunk 6]
```diff
@@ -240,6 +243,13 @@ defmodule CadetWeb.AssessmentsController do
               "The external library for this question"
             )
           end
+        end,
+      GraderInfo:
+        swagger_schema do
+          properties do
+            id(:integer, "grader id", required: true)
```
[the remark this answers is not in the document]
Comment (jiachen247) [reply lib/cadet_web/controllers/assessments_controller.ex]: I was thinking graders should know if anyone else has graded the submission previously? This is not reflected in the frontend as of yet.

[the remark this answers is not in the document]
Comment (jiachen247) [reply lib/cadet_web/controllers/assessments_controller.ex]: I was trying to standardize and be consistent with with how student is exposed.

Hmm is something just good to have in the frontend?

```
StudentInfo:
        swagger_schema do
          properties do
            id(:integer, "student id", required: true)
            name(:string, "student name", required: true)
          end
        end,
```

[the remark this answers is not in the document]
Comment (indocomsoft) [reply lib/cadet_web/controllers/assessments_controller.ex]: OK sure if you think so.

File: lib/cadet_web/controllers/assessments_controller.ex [hunk 7]
```diff
@@ -240,6 +243,13 @@ defmodule CadetWeb.AssessmentsController do
               "The external library for this question"
             )
           end
+        end,
+      GraderInfo:
```
Comment (indocomsoft) [inline lib/cadet_web/controllers/assessments_controller.ex]: You only need to define this once to keep things DRY (Don't Repeat Yourself). Perhaps the best place to put it is in the `grading_controller.ex`, so just remove this instance

File: lib/cadet_web/views/grading_view.ex [hunk 8]
```diff
@@ -43,14 +43,19 @@ defmodule CadetWeb.GradingView do
...
       maxXp: & &1.question.max_xp,
-      grade:
-        &transform_map_for_view(&1, %{
-          grade: :grade,
-          adjustment: :adjustment,
-          comment: :comment,
-          xp: :xp,
-          xpAdjustment: :xp_adjustment
-        })
+      grade: &build_grade(&1)
```
Comment (indocomsoft) [inline lib/cadet_web/views/grading_view.ex]: Prefer `&build_grade/1`

File: lib/cadet_web/views/view_helpers.ex [hunk 9]
```diff
@@ -3,8 +3,36 @@ defmodule CadetWeb.ViewHelpers do
...
+  end
+
+  def format_datetime(datetime = %DateTime{}) do
+    datetime
+    |> DateTime.truncate(:millisecond)
+    |> Timex.format!("{ISO:Extended}")
+  end
+
+  def format_datetime(datetime = %NaiveDateTime{}) do
+    timezone = Timezone.get("Asia/Singapore", Timex.now())
```
Comment (indocomsoft) [inline lib/cadet_web/views/view_helpers.ex]: Shit. I don't think this is correct. I think naive date time stores in UTC. Help me double check

**Pull request merged**